core/src/main/scala/kafka/log/OffsetMap.scala

```diff
@@ -68,90 +120,164 @@ class SkimpyOffsetMap(val memory: Int, val hashAlgorithm: String = "MD5") extend
 
   /**
    * The number of bytes of space each entry uses (the number of bytes in the hash plus an 8 byte offset)
+   * This evaluates to the number of bytes in the hash plus 8 bytes for the offset
+   * and, if applicable, another 8 bytes for non-offset compact strategy (set in the init method).
    */
-  val bytesPerEntry = hashSize + 8
-  
+  var bytesPerEntry = hashSize + longByteSize
```

Comment:
nit: I'd suggest we initialize this and the slots below as some sentinel value (e.g. -1) to indicate it is not initialized yet, instead of some "meaningful" value, to enforce that they can only be useful after `init` is called (again).